src/ApiConnectors/BaseApiConnector.php

```diff
@@ -42,21 +50,21 @@ protected function getProcessXmlService(): ProcessXmlService
     public function sendXmlDocument(\DOMDocument $document) {
         try {
             return $this->getProcessXmlService()->sendDocument($document);
-        } catch (\SoapFault $soapFault) {
+        } catch (\SoapFault | \ErrorException $exception) {
             /*
-             * Always reset the client. There may have been TCP connection issues, network issues, or logic issues on
-             * Twinfield's side, it won't hurt to get a fresh connection.
+             * Always reset the client. There may have been TCP connection issues, network issues,
+             * or logic issues on Twinfield's side, it won't hurt to get a fresh connection.
              */
             $this->connection->resetClient(Services::PROCESSXML());
 
-            if (stripos($soapFault->getMessage(), "Your logon credentials are not valid anymore. Try to log on again.") !== false) {
-                /*
-                 * Automatically retry and log on again.
-                 */
-                return $this->sendXmlDocument($document);
+            /* For a given set of exception messages, always retry the request. */
+            foreach (self::RETRY_REQUEST_EXCEPTION_MESSAGES as $message) {
+                if (stripos($exception->getMessage(), $message) !== false) {
+                    return $this->sendXmlDocument($document);
```

Comment:
This will create an infinite loop with infinite API calls if the error will be returned consistently. Can you change this to stop trying after 3 tries?